### PR TITLE
Feature/assessment v0 profile modal

### PR DESCRIPTION
**Pull request by bionara**
# Description

Modal for the profile intro. This is not wired up yet (left a todo).
Text font differs slightly but have left as-is, ie same as other modals.

Screenshot of how it looks:

image

Design https://www.figma.com/file/7TEcX7f6rw3Z0WYR4TpX3R/ZOE-COVID-Study-(App)---MASTER?node-id=17026%3A146197

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

## Screenshots

[comment]: <> (Please attach screenshots showing the change, if relevant)

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Checklist (for merger)

- [ ] Make sure the right merge strategy is chosen, commit merge into master and squash merge into develop.

### CODE REVIEW

File: assets/lang/en.json [hunk 1]
```diff
@@ -730,6 +730,12 @@
     "button": "Next",
     "button-end": "OK"
   },
+  "covid-profile-modal": {
+    "update": "Update",
+    "title": "New home for Vaccines",
+    "body": "Access and update your \nvaccine record from the new \n*Profile* tab",
```
Comment (juliatan) [inline assets/lang/en.json:736]: Don't think we should include the "\n"s

File: src/core/Analytics.ts [hunk 2]
```diff
@@ -153,6 +153,7 @@ enum EOtherEvents {
   SIGNUP = 'SIGNUP',
   VIEW_MODAL = 'VIEW_MODAL',
   VIEW_SCREEN = 'VIEW_SCREEN',
+  COVID_PROFILE_ONBOARDING_MODAL = 'COVID_PROFILE_ONBOARDING_MODAL',
```
Comment (juliatan) [inline src/core/Analytics.ts:156]: Nit pick - array alphabetically?

File: src/features/assessment/modals/ProfileOnboardingModal.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      track(events.COVID_PROFILE_ONBOARDING_MODAL);
+      setTracked(true);
+    }
+  });
+
+  const closeAndUpdateStartupInfo = React.useCallback(async () => {
+    // TODO when BE / API is set up in https://www.notion.so/joinzoe/Assessment-v0-backend-flags-fbc306ef537d4526ab0ba1dab4103e76
+
+    const infos: Partial<TPatientInfosRequest> = {
+      has_seen_covid_test_onboarding: true,
```
Comment (juliatan) [inline src/features/assessment/modals/ProfileOnboardingModal.tsx:38]: Should be `has_seen_assessment_flow_v0_onboarding`

File: src/features/assessment/modals/ProfileOnboardingModal.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    };
+    try {
+      await patientService.updatePatientInfo(props.patientId, infos);
+      props.onRequestClose();
+      dispatch(fetchStartupInfo());
+    } catch (_) {}
```
Comment (juliatan) [inline src/features/assessment/modals/ProfileOnboardingModal.tsx:44]: I think we should add error message state, or alternatively don't catch. We can then rely on `updatePatientInfo` global error handler instead that I set up earlier this week.

File: src/features/assessment/modals/ProfileOnboardingModal.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+import { TPatientInfosRequest } from '@covid/core/user/dto/UserAPIContracts';
+import i18n from '@covid/locale/i18n';
+import { sizes } from '@covid/themes';
+import { colors } from '@theme';
+import * as React from 'react';
+import { Image, StyleSheet, View } from 'react-native';
+import { useDispatch } from 'react-redux';
+
+interface IProps {
+  onPressClose?: () => void;
```
Comment (juliatan) [inline src/features/assessment/modals/ProfileOnboardingModal.tsx:15]: we can delete this.

File: src/features/assessment/modals/ProfileOnboardingModal.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  return (
+    <Modal
+      footerChildren={footerChildren}
+      modalName="ProfileOnboarding"
+      onRequestClose={closeAndUpdateStartupInfo}
+      testID="covid-profile-modal"
+      visible={props.visible}
+    >
+      <View style={styles.wrapper}>
+        <View style={styles.update}>
```
Comment (juliatan) [inline src/features/assessment/modals/ProfileOnboardingModal.tsx:62]: We have a `<Tag>` component if you want to use that instead.

File: src/features/assessment/modals/ProfileOnboardingModal.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    flex: 1,
+    height: 400,
+    resizeMode: 'contain',
+  },
+  text: {
+    marginBottom: sizes.m,
+    textAlign: 'center',
+  },
+  title: {
+    marginBottom: sizes.m,
```
Comment (juliatan) [inline src/features/assessment/modals/ProfileOnboardingModal.tsx:87]: Nitpick: marginVertical can be used to save a line.

File: src/features/assessment/modals/ProfileOnboardingModal.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+      {i18n.t('covid-profile-modal.button-end')}
+    </BrandedButton>
+  );
+
+  return (
+    <Modal
+      footerChildren={footerChildren}
+      modalName="ProfileOnboarding"
+      onRequestClose={closeAndUpdateStartupInfo}
+      testID="covid-profile-modal"
```
Comment (juliatan) [inline src/features/assessment/modals/ProfileOnboardingModal.tsx:58]: Not sure if you intended to call it "covid-profile-modal" - should it just be "profile-modal"?

File: src/features/assessment/modals/__tests__/ProfileOnboardingModal.test.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+describe('Profile onboarding modal tests', () => {
+  it('renders modal', async () => {
+    const onPress = jest.fn();
+    const { getByTestId } = render(
+      <ThemeProvider theme={theme}>
+        <ReduxProvider store={store}>
+          <SafeAreaProvider initialMetrics={initialMetrics}>
+            <ProfileOnboardingModal
+              visible
+              onPressClose={onPress}
```
Comment (juliatan) [inline src/features/assessment/modals/__tests__/ProfileOnboardingModal.test.tsx:45]: Can delete..

**Review by juliatan — COMMENTED** [review]
Nice going Ben!

File: assets/lang/en.json (hunk 1 above)
Comment (bionara) [reply assets/lang/en.json:736]: QA will pick up on the alignment, but have removed!

File: src/features/assessment/modals/ProfileOnboardingModal.tsx (hunk 3 above)
Comment (bionara) [reply src/features/assessment/modals/ProfileOnboardingModal.tsx:38]: sorry, left this as part of the todo above. have updated it now tho.

File: src/features/assessment/modals/ProfileOnboardingModal.tsx (hunk 4 above)
Comment (bionara) [reply src/features/assessment/modals/ProfileOnboardingModal.tsx:44]: will handle this in subsequent ticket (as part of todo above)

File: src/features/assessment/modals/ProfileOnboardingModal.tsx (hunk 5 above)
Comment (bionara) [reply src/features/assessment/modals/ProfileOnboardingModal.tsx:15]: done

File: src/features/assessment/modals/ProfileOnboardingModal.tsx (hunk 7 above)
Comment (bionara) [reply src/features/assessment/modals/ProfileOnboardingModal.tsx:87]: nice one, done

File: src/features/assessment/modals/ProfileOnboardingModal.tsx (hunk 8 above)
Comment (bionara) [reply src/features/assessment/modals/ProfileOnboardingModal.tsx:58]: have updated all instances of covid-profile-modal to profile-modal

File: src/features/assessment/modals/__tests__/ProfileOnboardingModal.test.tsx (hunk 9 above)
Comment (bionara) [reply src/features/assessment/modals/__tests__/ProfileOnboardingModal.test.tsx:45]: booted

**Review by juliatan — APPROVED** [review]

**Pull request merged**